### PR TITLE
feat: transaction export functionality

### PR DESCRIPTION
**Pull request by alexbarnsley**
## Summary

https://app.clickup.com/t/861mxjtqe

Requires https://github.com/ArdentHQ/arkscan/pull/360 to be merged first

todo:

- [x] tests
- [x] stan
- [x] toast alert on export
- [x] retry button on error

## Checklist

-   [ ] I checked my UI changes against the design and there are no notable differences
-   [ ] I checked my UI changes for any responsiveness issues
-   [ ] I checked my UI changes in light AND dark mode
-   [ ] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I opened a corresponding card on Clickup for any remaining TODOs in my code
-   [ ] I added a short description on how to test this PR _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

**Review by samharperpittam — CHANGES_REQUESTED** [review]
Slight issue
image

File: resources/js/export.js [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+        total: "Total [:networkCurrency]",
+        amountFiat: "Value [:userCurrency]",
+        feeFiat: "Fee [:userCurrency]",
+        totalFiat: "Total [:userCurrency]",
+        rate: "Rate [:userCurrency]",
+    };
+
+    const getTransactionAmount = (transaction) => {
+        let amount = transaction.amount / 1e8;
+        if (transaction.type === 6 && transaction.typeGroup === 1) {
```
Comment (ItsANameToo) [inline resources/js/export.js:33]:
```suggestion
        if (transaction.type === 6 && transaction.typeGroup === 1) {
```

might be nice to change this to enums. They exist in PSDK or in the JS crypto package iirc

File: resources/js/export.js [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+            return transaction.asset.payments.reduce(
+                (totalAmount, recipientData) => {
+                    if (recipientData.recipientId === address) {
+                        if (totalAmount < 0) {
+                            totalAmount = 0;
+                        }
+
+                        totalAmount += recipientData.amount / 1e8;
```
Comment (ItsANameToo) [inline resources/js/export.js:41]: change this to a method that takes arktoshi and changes it to a "regular" number, as it's reused quite a bit

File: resources/views/components/modals/export-transactions/export-status.blade.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
+<div class="flex flex-col">
+    <div class="flex justify-center mb-6">
+        <div x-show="exportStatus === 'PENDING_DOWNLOAD'">
```
Comment (ItsANameToo) [inline resources/views/components/modals/export-transactions/export-status.blade.php:3]: maybe an enum for this as well?

**Review by ItsANameToo — CHANGES_REQUESTED** [review]

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**